src/assets/wise5/services/createBranchService.ts

```diff
@@ -0,0 +1,154 @@
+import { Injectable } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { TeacherProjectService } from './teacherProjectService';
+import { DialogWithSpinnerComponent } from '../directives/dialog-with-spinner/dialog-with-spinner.component';
+import {
+  CHOICE_CHOSEN_VALUE,
+  SCORE_VALUE,
+  WORKGROUP_ID_VALUE
+} from '../../../app/domain/branchCriteria';
+import { CreateBranchParams } from '../common/CreateBranchParams';
+import { Transition } from '../common/Transition';
+import { TransitionCriteria } from '../common/TransitionCriteria';
+import { TransitionCriteriaParams } from '../common/TransitionCriteriaParams';
+
+@Injectable()
+export class CreateBranchService {
+  protected readonly CHOICE_CHOSEN_VALUE: string = CHOICE_CHOSEN_VALUE;
+  protected readonly SCORE_VALUE: string = SCORE_VALUE;
+
+  constructor(private dialog: MatDialog, private projectService: TeacherProjectService) {}
+
+  createBranch(params: CreateBranchParams): Promise<void> {
+    this.showCreatingBranchMessage();
+    const targetNode = this.projectService.getNodeById(params.branchStepId);
+    const nodeIdTargetNodeTransitionsTo =
+      targetNode.transitionLogic.transitions.length > 0
+        ? targetNode.transitionLogic.transitions[0].to
+        : '';
+    targetNode.transitionLogic.transitions = [];
+    const newNodeIds = this.createNewNodeIds(params.pathCount);
+    this.createPathSteps(params, targetNode, newNodeIds);
+    const mergeStep: any =
+      params.mergeStepId === ''
+        ? this.createMergeStep(newNodeIds, nodeIdTargetNodeTransitionsTo)
+        : this.projectService.getNodeById(params.mergeStepId);
+    this.setPathStepTransitions(newNodeIds, mergeStep.id);
+    this.setTargetNodeTransitionLogic(targetNode, params.criteria);
+    this.addNewNodeIdsToParentGroup(params.branchStepId, [...newNodeIds, mergeStep.id]);
+    return this.projectService.saveProject().then(() => {
+      this.hideCreatingBranchMessage();
+    });
+  }
+
+  private createNewNodeIds(pathCount: number): string[] {
+    const newNodeIds = [];
+    for (let i = 0; i < pathCount; i++) {
+      const newNodeId = this.projectService.getNextAvailableNodeId(newNodeIds);
+      newNodeIds.push(newNodeId);
```

Comment:
Write this in one line?
```suggestion
      newNodeIds.push(this.projectService.getNextAvailableNodeId(newNodeIds));
```